lib/dl_connector_bundle_chs3/dl_connector_bundle_chs3/chs3_base/core/us_connection.py

```diff
@@ -119,6 +122,12 @@ def s3_access_key_id(self) -> str:
     def s3_secret_access_key(self) -> str:
         return self._connector_settings.SECRET_ACCESS_KEY
 
+    def get_full_s3_filename(self, s3_filename_suffix: Optional[str]) -> Optional[str]:
+        if s3_filename_suffix is None:
+            return None
```

Comment:
Why `Optional` (both in arguments and in return type)? I can't see any calls where either can be None (except for tests)

Reply:
Right now `s3_filename_suffix` will be loaded as `None` for old connections until migration. 
After migration `Optional` should be removed (with this fallback https://github.com/datalens-tech/datalens-backend/blob/23b2d1e5d8b14eda297ee74519e5143414de1446/lib/dl_connector_bundle_chs3/dl_connector_bundle_chs3/chs3_base/core/data_source.py#L120-L124)

Reply:
Yes, exactly, it is tested against being None here in the line 120 before the `get_full_s3_filename` call

Reply:
okay, you are right, I can remove Optionals from the method. But actually it's just a copy of old property `src.s3_filename` signature that can be None if save task haven't succeeded yet.

Reply:
Just a reminder for all of us, that this check will have to be moved above the fallback after the latter is removed

https://github.com/datalens-tech/datalens-backend/blob/ad03ba8e11ff3eaba66a5065530e1df173c54772/lib/dl_connector_bundle_chs3/dl_connector_bundle_chs3/chs3_base/core/data_source.py#L124-L125